### PR TITLE
Defects beta connor

### PR DESCRIPTION
**Pull request by ConnorBoulais**
This PR fixes the defects assigned to me on Jira, as well as connects the task checkmarks on the Mission Overview screen to the backend

### CODE REVIEW

File: src/Components/Courses.js [hunk 1]
```diff
@@ -4,46 +4,60 @@ import { GET_COURSE } from '../gqlQueries.js';
...
-        "courseId": "a824af5ae39",
-        "course": "Math",
-        "instructor": "Mr. Butcher",
-        "description": "Mathematics Calc 1 - 4"
-      },
-      {
-        "courseId": "b2abdb129d3",
-        "course": "Physics",
-        "instructor": "Mr. Butcher",
-        "description": "Physics introductory course"
```
Comment (ryanmadamba) [inline src/Components/Courses.js:46]: Good getting rid of unused courses

File: src/App.css [hunk 2]
```diff
@@ -47,7 +47,7 @@ h2 {
     background-color: #4274F3;
     position:fixed;
     width: 100%;
-    z-index: 4;
+    z-index: 10;
```
Comment (tddumont) [inline src/App.css:50]: I had no idea that this property existed!

**Review by ryanmadamba — APPROVED** [review]
Good job fixing all of these defects and bugs!

File: src/Components/Courses.js [hunk 3]
```diff
@@ -4,46 +4,60 @@ import { GET_COURSE } from '../gqlQueries.js';
 import './CourseScreen.css';
 import { useHistory } from 'react-router';
 
+import Card from "@material-ui/core/Card";
```
Comment (tddumont) [inline src/Components/Courses.js:7]: I like the look of the new course card!

File: src/Components/Goals/GoalForm.js [hunk 4]
```diff
@@ -46,31 +48,36 @@ function GoalForm(props) {
...
     }).then(() => props.closeFormCallBack());
+    setSubmitButtonEnable(false);
   }
 
   function handleAddSubgoal() {
     if(subTitle == '' || subDate == ''){
       alert('All fields are required to create a sub goal')
     }
+    else if(!subDate || !(subDate._isValid)){
+      alert('Sub Goal Due Date is Invalid')
```
Comment (tddumont) [inline src/Components/Goals/GoalForm.js:66]: Great to address this edge case!

File: src/Components/Task.js [hunk 5]
```diff
@@ -142,7 +158,7 @@ function Task(props) {
     }).catch((error) => {
       console.log(error)
       if(error.message.includes("ineligible for submission")){
-        alert('Task ineliglbe for submission, please answer all questions')
+        alert('Task ineligible for submission, please answer all questions')
```
Comment (tddumont) [inline src/Components/Task.js:161]: Good find!

**Review by tddumont — APPROVED** [review]
Lots of great work!

**Pull request merged**